### PR TITLE
[serdes] @cached_property all ObjectSerializer.constructor_param_names

### PR DESCRIPTION
**Pull request by alangenfeld**
now that we have Pydantic models in use we can see that this method is not cheap so is worth memoizing

## How I Tested These Changes

bk

### CODE REVIEW

**Comment by alangenfeld** [comment]
* **#21553** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21553?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alangenfeld and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by alangenfeld** [comment]
think 
https://github.com/dagster-io/dagster/pull/21432
actually solved the underlying problem

**Comment by alangenfeld** [comment]
never mind seems costly still

File: python_modules/dagster/dagster/_serdes/serdes.py [hunk 1]
```diff
@@ -689,9 +687,12 @@ def object_as_mapping(self, value: T_PydanticModel) -> Mapping[str, Any]:
 
         return result
 
-    @property
     def constructor_param_names(self) -> Sequence[str]:
-        return [field.alias or key for key, field in model_fields(self.klass).items()]
+        return [field.alias or key for key, field in self._model_fields.items()]
+
+    @cached_property
+    def _model_fields(self) -> Mapping[str, ModelFieldCompat]:
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_serdes/serdes.py:694]: cache at this layer instead since its used in 2 places

**Review by sryza — APPROVED** [review]

**Pull request merged**